connect/api/src/main/java/org/apache/kafka/connect/sink/SinkRecord.java

```diff
@@ -18,55 +18,74 @@
 package org.apache.kafka.connect.sink;
 
 import org.apache.kafka.common.annotation.InterfaceStability;
+import org.apache.kafka.common.record.TimestampType;
 import org.apache.kafka.connect.connector.ConnectRecord;
 import org.apache.kafka.connect.data.Schema;
 
 /**
  * SinkRecord is a {@link ConnectRecord} that has been read from Kafka and includes the kafkaOffset of
  * the record in the Kafka topic-partition in addition to the standard fields. This information
  * should be used by the SinkTask to coordinate kafkaOffset commits.
+ *
+ * It also includes the {@link TimestampType}, which may be {@link TimestampType#NO_TIMESTAMP_TYPE}, and the relevant
+ * timestamp.
  */
 @InterfaceStability.Unstable
 public class SinkRecord extends ConnectRecord {
     private final long kafkaOffset;
+    private final TimestampType timestampType;
 
-    public SinkRecord(String topic, int partition, Schema keySchema, Object key, Schema valueSchema, Object value, long kafkaOffset) {
-        super(topic, partition, keySchema, key, valueSchema, value);
+    public SinkRecord(String topic, int partition,
+                      Schema keySchema, Object key,
+                      Schema valueSchema, Object value,
+                      long kafkaOffset, long timestamp, TimestampType timestampType) {
+        super(topic, partition, keySchema, key, valueSchema, value, timestamp);
         this.kafkaOffset = kafkaOffset;
+        this.timestampType = timestampType;
     }
 
     public long kafkaOffset() {
         return kafkaOffset;
     }
 
+    public TimestampType timestampType() {
+        return timestampType;
+    }
+
     @Override
     public boolean equals(Object o) {
-        if (this == o)
+        if (this == o) {
```

Comment:
I think your IDE is adding extra braces. (I know this is a bit annoying between Apache Kafka & Confluent projects since they use different styling. iirc default IntelliJ styling works for Kafka if you are using IntelliJ)